tests/system/robot/notepadTests.py

```diff
@@ -0,0 +1,39 @@
+# A part of NonVisual Desktop Access (NVDA)
+# Copyright (C) 2021 NV Access Limited
+# This file may be used under the terms of the GNU General Public License, version 2 or later.
+# For more details see: https://www.gnu.org/licenses/gpl-2.0.html
+
+"""Logic for reading text using NVDA in the notepad text editor.
+"""
+# imported methods start with underscore (_) so they don't get imported into robot files as keywords
+from SystemTestSpy import (
+	_getLib,
+)
+
+# Imported for type information
+from NotepadLib import NotepadLib as _NotepadLib
+from AssertsLib import AssertsLib as _AssertsLib
+
+_notepad: _NotepadLib = _getLib("NotepadLib")
+_asserts: _AssertsLib = _getLib("AssertsLib")
+
+
+def test_symbolLevelWord(isSymbolLevelWordAllExpected=True):
+	wordsWithSymbols = ['He', 'said', '(quietly),', '"Hello,', 'Jim".']
+	symbolMap = {
+		'(': 'left paren',
+		')': 'right paren',
+		',': 'comma',
+		'"': 'quote',
+		'.': 'dot',
+	}
+	textStr = ' '.join(wordsWithSymbols)
+	_notepad.prepareNotepad(f"Test: {textStr}")
+	for expectedWord in wordsWithSymbols:
+		wordSpoken = _notepad.getSpeechAfterKey("numpad6")
+		for symbol in symbolMap.keys():
+			if isSymbolLevelWordAllExpected:
+				expectedWord = expectedWord.replace(symbol, f" {symbolMap[symbol]}{symbol}")
+		# unlike other symbols used, symbols.dic doesn't preserve quote symbols with SYMPRES_ALWAYS
```

Comment:
Logic in tests smells, being pragmatic it may not easily be avoidable. The problems it introduces are:
- higher possibility of a bug in the test itself.
- the test is harder to read, it doesn't serve as documentation for the expected behavior as well.

Parameterized tests are useful, but mostly when there is a lot of data to get through. Say every combination of a number of parameters. If it's a small number of combinations personally I think explicit is better.

You could instead have something like: 

``` python
wordsToExpectedSymbolSpeech = {
'Say':'Say',
'(quietly)': 'left paren  quietly  right paren',
'"Hello,': 'quote  Hello  comma',
'Jim".': 'Jim  dot  quote',
}
def test_symbolLevelWord_symbolLevelWord:
	textStr = ' '.join(wordsToExpectedSymbolSpeech.keys())
	_notepad.prepareNotepad(f"Test: {textStr}")
	for expectedWord in wordsToExpectedSymbolSpeech.values():
		wordSpoken = _notepad.getSpeechAfterKey("numpad6")  # navigate to next word
		# unlike other symbols used, symbols.dic doesn't preserve quote symbols with SYMPRES_ALWAYS
		expectedWord = expectedWord.replace('"', ' ').strip()
		_asserts.strings_match(wordSpoken, expectedWord)

wordsToExpectedSymbolSpeech = {
'Say':'Say',
'(quietly)': 'quietly',
'"Hello,': 'Hello',
'Jim".': 'Jim',
}
def test_symbolLevelWord_symbolLevelSome:  # todo: is 'some' the correct level?
	textStr = ' '.join(wordsToExpectedSymbolSpeech.keys())
	_notepad.prepareNotepad(f"Test: {textStr}")
	for expectedWord in wordsToExpectedSymbolSpeech.values():
		wordSpoken = _notepad.getSpeechAfterKey("numpad6")  # navigate to next word
		# unlike other symbols used, symbols.dic doesn't preserve quote symbols with SYMPRES_ALWAYS
		expectedWord = expectedWord.replace('"', ' ').strip()
		_asserts.strings_match(wordSpoken, expectedWord)
```
```

Reply:
I prefer that approach too, I've updated the tests to follow this pattern, but I have made some changes to simplify it further, if you want to take another look.

Reply:
Looks good to me.